openmdao/components/tests/test_implicit_func_comp.py

```diff
@@ -0,0 +1,233 @@
+
+        f = (omf.wrap(apply_nonlinear)
+                .add_input('z', val=np.array([-1., -1.]))
+                .add_input('x', val=2.)
+                .add_output('y1', resid='r_y1')
+                .add_output('y2', resid='r_y2')
+                .add_output('R_y1', resid='r_R_y1')
+                .add_output('R_y2', resid='r_R_y1')
+                .declare_partials('y1', 'y1')
+                .declare_partials('y2', ['z', 'y1', 'y2'])
+                .declare_partials('R_y1', ['R_y1', 'x', 'z', 'y1', 'y2'])
+                .declare_partials('R_y2', ['R_y2','z', 'y1', 'y2']))
+
+        def linearize(z, x, y1, y2, R_y1, R_y2, J):
+
+            J['y1', 'y1'] = -1.
+            J['R_y1','R_y1'] = -1
+
+            J['R_y1', 'x'] = [1]
+            J['R_y1', 'z'] = [2*z[0], 1]
+            J['R_y1', 'y1'] = -1.
+            J['R_y1', 'y2'] = -0.2
+
+            J['y2','y2'] = -1
+
+            J['R_y2','R_y2'] = -1
+            J['y2','z'] = [1, 1]
+            J['y2','y1'] = 0.5*y1**-0.5
+
+            J['R_y2','y2'] = -1
+            J['R_y2','z'] = [1, 1]
+            J['R_y2','y1'] = 0.5*y1**-0.5
+
+        p_opt = om.Problem()
+
+        p_opt.model = om.ImplicitFuncComp(f, linearize=linearize) #, solve_nonlinear=solve_nonlinear)
+
+        newton = p_opt.model.nonlinear_solver = om.NewtonSolver(solve_subsystems=False)
+        newton.options['iprint'] = 0
+        p_opt.model.linear_solver = om.DirectSolver(assemble_jac=True)
+
+        p_opt.driver = om.ScipyOptimizeDriver()
+        p_opt.driver.options['disp'] = False
+
+        p_opt.model.add_design_var('y1', lower=-10, upper=10)
+        p_opt.model.add_constraint('R_y1', equals=0)
+
+        # this objective doesn't really matter... just need something there
+        p_opt.model.add_objective('y2')
+
+        p_opt.setup()
+
+        # set
+        p_opt['y2'] = 5
+        p_opt['y1'] = 5
+
+        p_opt.run_driver()
+
+        np.testing.assert_almost_equal(p_opt['y1'], 2.109516506074582, decimal=5)
+        np.testing.assert_almost_equal(p_opt['y2'], -0.5475825303740725, decimal=5)
+        np.testing.assert_almost_equal(p_opt['x'], 2.0, decimal=5)
+        np.testing.assert_almost_equal(p_opt['z'], np.array([-1., -1.]), decimal=5)
+
+    def test_apply_nonlinear(self):
+
+        def apply_nl(a, b, c, x):
+            R_x = a * x ** 2 + b * x + c
+            return R_x
+
+        f = (omf.wrap(apply_nl)
+                .add_output('x', resid='R_x', val=0.0)
+                .declare_partials(of='*', wrt='*', method='cs')
+                )
+
+        p = om.Problem()
+        p.model.add_subsystem('comp', om.ImplicitFuncComp(f))
+
+        # need this since comp is implicit and doesn't have a solve_linear
+        p.model.linear_solver = om.DirectSolver()
+        newton = p.model.nonlinear_solver = om.NewtonSolver(solve_subsystems=False)
+        newton.options['iprint'] = 0
```

Comment:
couldn't you do this in one line by passing `iprint` as an arg as well?
`p.model.nonlinear_solver = om.NewtonSolver(solve_subsystems=False, iprint=0)`

Reply:
fixed